documentation/PreferencesServer.md

```diff
@@ -350,6 +350,434 @@ There are two important things to note here:
+suitable message will be returned.
+
+```json
+{
+    "isError": true,
+    "errorCode": "APPROPRIATE_ERROR_CODE",
+    "message": "System error described here."
+}
+```
+
+### GET /prefssafes
+
+Returns a list of preferences safes, including only basic information about each one. Ideal for building a table
+or listing of preferences safes. Each item in the list representing a preferences safe will include `id`, `name`,
+`email`, `created`, and `updated`. This endpoint will likely have more options in the future, such as sorting,
+paging, etc.
+
+#### Example GET
+
+URL: `http://preferences.gpii.net/prefssafes`
+
+Example return payload:
+
+```json
+{
+    "total_rows": 2,
+    "offset": 0,
+    "rows": [
+        {
+            "name": "Alice",
+            "email": "alice@gpii.net",
+            "created": "2017-12-14T19:55:11.640Z",
+            "updated": null,
+            "id": "prefsSafe-1"
+        },
+        {
+            "name": null,
+            "email": null,
+            "created": "2017-12-14T19:55:11.640Z",
+            "updated": null,
+            "id": "prefsSafe-2"
+        }
+    ]
+}
+```
+
+There should never be an error payload for this endpoint. In the situation where there are no preferences
+safes stored in the system it will merely contain zero rows.
+
+### GET /prefssafe-keys/:prefsSafeId
+
+This will return the associated keys and credentials documents for a given preferences safe, in a `rows` field.
+Also included is a `total_rows` and `offset` field, but do note that the `total_rows` field is not accurate as
+of the time of writing, and should be ignored.
+
+#### Example GET
+
+URL: `http://preferences.gpii.net/prefssafe-keys/prefsSafe-alice`
+
+An example payload for a particlar safe may be:
+
+```json
+{
+    "total_rows": 2,
+    "offset": 0,
+    "rows": [
+        {
+            "id": "8f3085a7-b65b-4648-9a78-8ac7de766997",
+            "type": "gpiiCloudSafeCredential",
+            "schemaVersion": "0.1",
+            "prefsSafeId": "prefsSafe-alice",
+            "gpiiExpressUserId": "org.couch.db.user:alice"
+        },
+        {
+            "id": "57A68E84-03A9-4ADD-9365-11C75E4F1B0E",
+            "type": "gpiiKey",
+            "schemaVersion": "0.1",
+            "prefsSafeId": "prefsSafe-alice",
+            "prefsSetId": "gpii-default",
+            "revoked": false,
+            "revokedReason": null,
+            "timestampCreated": "2017-12-14T19:55:11.640Z",
+            "timestampUpdated": null
+        }
+    ]
+}
+```
+
+### POST /prefssafe-key-create
+
+This endpoint will add a new `gpiiKey` document to the system. There are no URL parameters, but the JSON body takes
+3 fields, one of them optional. Required are the `prefsSafeId` and `prefsSetId`. These indicate the preferences safe,
+and the respective preferences set that this key and token will operate upon. Optionally you can pass in a unique

```

Comment:
"this key and token" -> "this GPII key". All appearances of "user token", "token" that reference a GPII key should be replaced with "GPII key". There had been a round of terminology sync in this repo a few months ago. This comment applies to the entire pull request.

Reply:
updated